src/main/java/ecdar/backend/BackendDriver.java

```diff
@@ -48,18 +48,32 @@ public void addRequestToExecutionQueue(GrpcRequest request) {
         requestQueue.add(request);
     }
 
-    public void setConnectionAsAvailable(EngineConnection engineConnection) {
-        var relatedQueue = this.availableEngineConnections.get(engineConnection.getEngine());
-        if (!relatedQueue.contains(engineConnection)) relatedQueue.add(engineConnection);
+    /**
+     * Signal that the EngineConnection can be used not in use and available for queries
+     *
+     * @param connection to make available
+     */
+    public void setConnectionAsAvailable(EngineConnection connection) {
+        var relatedQueue = this.availableEngineConnections.get(connection.getEngine());
+        if (!relatedQueue.contains(connection)) relatedQueue.add(connection);
     }
 
     /**
      * Close all open engine connection and kill all locally running processes
-     *
-     * @throws IOException if any of the sockets do not respond
      */
-    public void closeAllEngineConnections() throws IOException {
+    public void closeAllEngineConnections() {
         for (EngineConnection ec : startedEngineConnections) ec.close();
+        startedEngineConnections.clear();
+        availableEngineConnections.clear();
+    }
+
+    /**
+     * Close all engine connections and stop all queries
+     */
+    public void reset() {
+        closeAllEngineConnections();
+        BackendHelper.stopQueries();
+        requestQueue.clear();
```

Comment:
I think this need a try catch block and we should consider what would happen if ``closeAllEngineConnections`` fails? Also maybe we should just call ``stopQueries`` for the backend we are closing and not all when possible only some of the connections were closed.

Reply:
There are currently no exceptions to try/catch in the related logic. I could add some logic to the `channel.shutdown` and `process.destroy()` in `EngineConnection::close()` to verify the shutdown? That might be the ideal approach to insure the successful shutdown of both channels and processes.
I will implement this shortly :+1:

Reply:
Regarding the `reset()` method, then I believe the expected effect would be a complete reset of the backend